src/app/tide/compiler.h

```diff
@@ -116,39 +116,14 @@ class TheoreticalPeakCompiler {
     return pos_; 
   }
 
-  // The following three functions assume peaks unsorted, and check that each peak
-  // doesn't point past end of cache.
+  // The following three functions assume peaks unsorted;
+  // It does not check that each peak doesn't point past 
+  // end of cache. Anymore. Modification included on 2021 Dec, by AKF
```

Comment:
How much of a speed boost do you get when removing this code? It seems dangerous to not check the cache. I guess the (probably reasonable) assumption is that this doesn't happen very often?

Reply:
It is check before this function call. In the AddYIon and AddBIon functions in theoretical_peak_set.h:93 and 107 this condition is checked, see lines 102 and 116 in theoretical_peak_set.h
Theoretical peaks beyond the cache end are not kept.